context/app/static/js/components/Header/UserLinks/UserLinks.spec.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import { render, screen } from 'test-utils/functions';
+import userEvent from '@testing-library/user-event';
+
+import UserLinks from './UserLinks';
+
+test('should be "User Profile" when not authenticated', () => {
+  render(<UserLinks isAuthenticated={false} />);
+  expect(screen.getByText('User Profile')).toBeInTheDocument();
+  userEvent.click(screen.getByText('User Profile'));
+  // In drop-down:
+  expect(screen.getByText('My Lists')).toBeInTheDocument();
```

Comment:
In older tests, I haven't been perfect about this, but you need to `await` one of the appearing elements after firing an event. See the testing recipes in the testing-library link in the other comment.

Here's a snippet from the example.
```
  fireEvent.click(screen.getByText('Load Greeting'))

  await waitFor(() => screen.getByRole('heading'))
```